src/extra/AutoUi/Collection/index.tsx

```diff
@@ -266,20 +274,23 @@ export const AutoUICollection = <T extends AutoUIBaseResource<T>>({
 				setIsBusyMessage(undefined);
 			}
 		},
-		[sdk?.tags, refresh],
+		[sdk?.tags, refresh, selected, filteredSelected],
 	);
 
 	const onActionTriggered = React.useCallback((actionData: ActionData<T>) => {
 		setActionData(actionData);
 	}, []);
 
-	const onActionDone = React.useCallback((isSuccessful: boolean) => {
-		if (isSuccessful) {
-			setSelected([]);
-		}
+	const onActionDone = React.useCallback(
+		(isSuccessful: boolean) => {
+			if (isSuccessful) {
+				setSelected(difference(selected, filteredSelected));
+			}
```

Comment:
If we approve to keep the selection after the action, we can just remove this piece of code.